From b6ef870ec77b50d4d28082e3015430aec5a5426e Mon Sep 17 00:00:00 2001
From: "Lu.Jiang" <lu.jiang@windriver.com>
Date: Wed, 10 Sep 2014 14:17:10 +0800
Subject: [PATCH 357/987] Update the operation mapping settings required by the
 DSP side for Maple and DMA.

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

For the DSP side the basic requirement for stashing is to prevent device
initiated reads and writes going to the DDR. RWNITC and WWSOT operation
mappings aid in achieving these objectives.

Signed-off-by: Varun Sethi <Varun.Sethi@freescale.com>
Change-Id: I1634ff877d9c8e7c6a3de95aee4a3848cc149973
Reviewed-on: http://git.am.freescale.net:8181/4128
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
Tested-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
Signed-off-by: Lu.Jiang <lu.jiang@windriver.com>
---
 drivers/iommu/fsl_pamu.c | 26 ++++++--------------------
 include/linux/iommu.h    |  4 +---
 2 files changed, 7 insertions(+), 23 deletions(-)

diff --git a/drivers/iommu/fsl_pamu.c b/drivers/iommu/fsl_pamu.c
index 48f0410..6d73e6d 100644
--- a/drivers/iommu/fsl_pamu.c
+++ b/drivers/iommu/fsl_pamu.c
@@ -764,26 +764,12 @@ static void __init setup_omt(struct ome *omt)
 	ome->moe[IOE_DIRECT0_IDX] = EOE_LDEC | EOE_VALID;
 	ome->moe[IOE_DIRECT1_IDX] = EOE_LDEC | EOE_VALID;
 
-	/* Configure OMI_DMA */
-	ome = &omt[OMI_DMA];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
-
-	/* Configure OMI_DMA_READI */
-	ome = &omt[OMI_DMA_READI];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_READI;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_READI;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
-
-	/* Configure OMI_MAPLE */
-	ome = &omt[OMI_MAPLE];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
+	/* Configure OMI_DSP */
+	ome = &omt[OMI_DSP];
+	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RWNITC;
+	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RWNITC;
+	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSAO;
+	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSAO;
 }
 
 /*
diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index e11b045..6cbe9ed 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -102,9 +102,7 @@ enum omap_index {
 	OMI_QMAN_PRIV,
 	OMI_CAAM,
 	OMI_PMAN,
-	OMI_DMA,
-	OMI_DMA_READI,
-	OMI_MAPLE,
+	OMI_DSP,
 	OMI_MAX,
 };
 
-- 
1.9.1

